src/main/java/io/wispforest/owo/serialization/impl/MapCodeck.java

```diff
@@ -0,0 +1,57 @@
+package io.wispforest.owo.serialization.impl;
+
+import io.wispforest.owo.serialization.Codeck;
+import io.wispforest.owo.serialization.Deserializer;
+import io.wispforest.owo.serialization.Serializer;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+public class MapCodeck<K, V> implements Codeck<Map<K, V>> {
+
+    private Supplier<Map<K, V>> mapConstructor = LinkedHashMap::new;
+
+    private final Codeck<V> codeck;
+
+    public Function<K, String> fromKey;
+    public Function<String, K> toKey;
+
+    private MapCodeck(Codeck<V> codeck, Function<K, String> fromKey, Function<String, K> toKey){
+        this.codeck = codeck;
+
+        this.fromKey = fromKey;
+        this.toKey = toKey;
+    }
+
+    public MapCodeck<K, V> mapConstructor(Supplier<Map<K, V>> mapConstructor){
+        this.mapConstructor = mapConstructor;
+
+        return this;
+    }
+
+    public static <V> MapCodeck<String, V> of(Codeck<V> codeck){
+        return new MapCodeck<>(codeck, s -> s, s -> s);
+    }
+
+    public <R> MapCodeck<R, V> keyThen(Function<K, R> toFunc, Function<R, K> fromFunc) {
```

Comment:
Is this method in the Public API? IntelliJ might complain about using it outside of owo if the impl package is made internal

Reply:
Possible, such is still up in the air about how the folder structure should be done.